src/gleam_stdlib.erl

```diff
@@ -127,3 +127,15 @@ string_contains(Haystack, Needle) ->
     _ ->
       true
   end.
+
+string_starts_with(_, "") -> true;
```

Comment:
Does this ever match? `""` is a charlist

Reply:
I'm sure you're right, I'll have to read up on that, still being an Erlang/Elixir newb.

Reply:
Ok, I understand this concept now. It did not ever match indeed.
I am pattern matching against an empty binary now, this does match in the required cases.

```erlang
 string_starts_with(_, <<>>) -> true;
```